src/Command/Diff.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+/*
+ * This file is part of the Manala package.
+ *
+ * (c) Manala <contact@manala.io>
+ *
+ * For the full copyright and license information, please refer to the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Manala\Command;
+
+use Manala\Env\EnvEnum;
+use Manala\Exception\HandlingFailureException;
+use Manala\Handler\Diff as DiffHandler;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\ConsoleOutput;
+use Symfony\Component\Console\Output\NullOutput;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+/**
+ * @author Maxime STEINHAUSSER <maxime.steinhausser@gmail.com>
+ */
+class Diff extends Command
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('diff')
+            ->setDescription('Computes the diff between the current project and the Manala templates.')
+            ->addArgument('cwd', InputArgument::OPTIONAL, 'The path of the application', getcwd())
+            ->addOption('env', null, InputOption::VALUE_OPTIONAL, 'One of the supported environment types', 'symfony');
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $cwd = realpath($input->getArgument('cwd'));
+        $envType = EnvEnum::create($input->getOption('env'));
+
+        if (!is_dir($cwd)) {
+            throw new \RuntimeException(sprintf('The working directory "%s" doesn\'t exist.', $cwd));
+        }
+
+        $handler = new DiffHandler($envType, $cwd, $output->isDecorated());
+        $verbosity = $output->getVerbosity();
+
+        try {
+            $output->setVerbosity(OutputInterface::VERBOSITY_NORMAL);
+
+            $handler->handle(function ($type, $buffer) use ($output) {
+                $output->write($buffer);
+            });
+
+            $output->setVerbosity($verbosity);
+        } catch (HandlingFailureException $e) {
+            $output->setVerbosity($verbosity);
+
+            $io = new SymfonyStyle($input, $output);
+            $io->error(['An error occurred during the process execution:', $handler->getErrorOutput()]);
+
+            return $handler->getExitCode();
+        }
+
+        $errIo = $this->getErrIo($input, $output);
+
+        if (!$handler->hasDiff()) {
+            $errIo->success('No diff found.');
+
+            return 0;
+        }
+
+        $errIo->note(<<<'EOTXT'
+You can apply the patch by executing the following command in the project working directory:
+
+$ patch -p1 < my_patch.patch
```

Comment:
How is set the patch filename? Is it `my_patch.patch`? I miss something ^^
Also imho we should be more explicit about that the project configuration is outdated, maybe something like:
```
Your project configuration is outdated, you can upgrade it by applying the patch on your project:

$ patch -p1 < my_patch.patch
```